cnd/src/config.rs

```diff
@@ -144,8 +158,16 @@ impl Ethereum {
 
         let chain_id = ethereum.chain_id;
         let geth = ethereum.geth.unwrap_or_else(Geth::new);
+        let tokens = ethereum.tokens.map_or_else(
+            || Tokens::from_config_or_chain_id(None, chain_id),
+            |tokens| Tokens::from_config_or_chain_id(tokens.dai, chain_id),
+        )?;
```

Comment:
Ha! Took me a while to figure out this pattern but its kinda nice :)

Reply:
Yeah, there is a lot going on here but I like how concise it is, especially with the error handling as well :)

Reply:
One unfortunate thing is that the arguments are in a weird order. The first one is the default `or_else` and the second one is the `map`. There is a reason though: https://github.com/rust-lang/rust/issues/53268#issuecomment-429988655

Reply:
This actually made me notice a certain inconsistency in how these functions are named :)